main.js

```diff
@@ -28,6 +30,22 @@ function createWindow () {
         // when you should delete the corresponding element.
         mainWindow = null;
     });
+    // Save current layout for layout mode
+    ipcMain.on('layout-changed', function(event, layout) {
+        if(process.argv.includes('--layout')) {
+            fs.writeFile('./current_layout.json',
+                JSON.stringify(layout.map(function(widget) {
+                    return {
+                        i: widget.i,
+                        x: widget.x,
+                        y: widget.y,
+                        w: widget.w,
+                        h: widget.h
+                    }
+                }), undefined, 4)
```

Comment:
Apologies for not doing any automated linting, but please add any missing semicolons.